### PR TITLE
[GUI.Qt] Move default format setup before application creation

### PR DESCRIPTION
**Pull request by alxbilger**
To fix rendering bug reported in https://github.com/sofa-framework/sofa/discussions/3100
Also restore the `msaa` command line option

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp [hunk 1]
```diff
@@ -226,8 +227,23 @@ class RealGUIFileListener : public sofa::helper::system::FileEventListener
 
 //======================= STATIC METHODS ========================= {
 
+void RealGUI::setupSurfaceFormat()
+{
+    if(!SOFA_GUI_QT_ENABLE_VSYNC)
+    {
+        static std::once_flag flag;
```
Comment (fredroy) [inline Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp]: I did not know about that 👌

File: Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp [hunk 2]
```diff
@@ -226,8 +227,23 @@ class RealGUIFileListener : public sofa::helper::system::FileEventListener
...
+        std::call_once(flag, []
+        {
+            QSurfaceFormat format;
+            format.setSwapInterval(0);
+            QSurfaceFormat::setDefaultFormat(format);
```
Comment (fredroy) [inline Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp]: Should we not set here the  compatibility profile and so on ?
(and the MSAA sampling here but we dont have the arg here unfortunately)

Comment (alxbilger) [reply Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp]: Compatibility profile is set only in QtViewer, right?
QtViewer setups a new `QSurfaceFormat` based on the default format, which was modified previously to support Vsync (or not). So, I guess that for QtViewer the compatibility profile is set properly. What do you think?
Do you think all this setup must go in the default QSurfaceFormat?

Comment (fredroy) [reply Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp]: Humm I was thinking just to uniformize/factorize/gather in one location the surfaceFormat initialization (both qt viewers would share the same surfaceformat once for all)

Comment (alxbilger) [reply Sofa/GUI/Qt/src/sofa/gui/qt/RealGUI.cpp]: Done. Let me know if it matches your thoughts

**Pull request merged**